### PR TITLE
consolidate chains info

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes BX-####
Figma link (if any):

moving code around to be better prepared when backend sends us chains

- references/chains has now all chains info, but native assets
- utils/chains has all chain related methods, i removed some of them since they were not necessary
- changed places of wagmi util methods to clientRpc now wagmi/index has only wagmi constructor methods
- used viem chains wherever is possible

### CODE REVIEW

File: src/entries/popup/pages/home/NFTs/NFTDetails.tsx [hunk 1]
```diff
@@ -804,9 +805,11 @@ const NFTAccordionAboutSection = ({
...
             <Inline alignVertical="center" space="6px">
               <ChainBadge
-                chainId={chainIdFromChainName(
-                  nft?.network || ChainName.mainnet,
-                )}
+                chainId={
+                  chainNameToIdMapping[
+                    (nft?.network as ChainName) || ChainName.mainnet
+                  ]
+                }
```
Comment (magiziz) [inline src/entries/popup/pages/home/NFTs/NFTDetails.tsx:812]: I assume doing `as` type assertion won't be an issue here right ?

File: src/entries/popup/pages/send/index.tsx [hunk 2]
```diff
@@ -283,14 +283,14 @@ export function Send() {
...
+      chainId: chainNameToIdMapping[nft.network],
       chainName: nft.network,
       isNativeAsset: false,
       name: nft.name,
       symbol: nft.collection.name,
-      uniqueId: `${nft.asset_contract.address || ''}_${chainIdFromChainName(
-        nft.network,
-      )}`,
+      uniqueId: `${nft.asset_contract.address || ''}_${
+        chainNameToIdMapping[nft.network as ChainName]
```
Comment (magiziz) [inline src/entries/popup/pages/send/index.tsx:292]: Same here ?

**Review by magiziz — APPROVED** [review]
Looks good to me. Just had a few questions.

File: src/entries/popup/pages/home/NFTs/NFTDetails.tsx (hunk 1 above)
Comment (estebanmino) [reply src/entries/popup/pages/home/NFTs/NFTDetails.tsx:812]: yeah but now that you mention i'll take another look to return that type from the nft interface to make it cleaner

Comment (estebanmino) [reply src/entries/popup/pages/home/NFTs/NFTDetails.tsx:812]: ok, this wasn't needed at all lol i updated

**Review by derHowie — APPROVED** [review]
lgtm 🚀

**Pull request merged**